chainercv/experimental/links/model/fcis/fcis_resnet101.py

```diff
@@ -0,0 +1,234 @@
+import chainer.functions as F
+import chainer.links as L
+import numpy as np
+
+from chainercv.experimental.links.model.fcis import FCIS
+from chainercv.functions import psroi_pooling_2d
+from chainercv.links import Conv2DBNActiv
+from chainercv.links.model.faster_rcnn.region_proposal_network import \
+    RegionProposalNetwork
+from chainercv.links.model.faster_rcnn.utils.loc2bbox import loc2bbox
+from chainercv.links.model.resnet.resblock import ResBlock
+from chainercv.utils import download_model
+
+
+class FCISResNet101(FCIS):
+
+    _models = {
+        'sbd': {
+            'n_fg_class': 20,
+            'url': 'https://github.com/yuyu2172/share-weights/releases/'
+            'download/0.0.6/fcis_resnet101_2018_04_14.npz'
+        }
+    }
+    feat_stride = 16
+
+    def __init__(
+            self,
+            n_fg_class=None,
+            pretrained_model=None,
+            min_size=600, max_size=1000,
+            ratios=[0.5, 1, 2], anchor_scales=[8, 16, 32],
+            rpn_initialW=None,
+            loc_normalize_mean=(0.0, 0.0, 0.0, 0.0),
+            loc_normalize_std=(0.2, 0.2, 0.5, 0.5),
+            iter2=True,
+            proposal_creator_params={
+                'nms_thresh': 0.7,
+                'n_train_pre_nms': 6000,
+                'n_train_post_nms': 300,
+                'n_test_pre_nms': 6000,
+                'n_test_post_nms': 300,
+                'force_cpu_nms': False,
+                'min_size': 16
+            }
+    ):
+        if n_fg_class is None:
+            if pretrained_model not in self._models:
+                raise ValueError(
+                    'The n_fg_class needs to be supplied as an argument')
+            n_fg_class = self._models[pretrained_model]['n_fg_class']
+
+        if rpn_initialW is None:
+            rpn_initialW = chainer.initializers.Normal(0.01)
+
+        extractor = ResNet101Extractor()
+        rpn = RegionProposalNetwork(
+            1024, 512,
+            ratios=ratios,
+            anchor_scales=anchor_scales,
+            feat_stride=self.feat_stride,
+            initialW=rpn_initialW,
+            proposal_creator_params=proposal_creator_params
+        )
+        head = FCISResNet101Head(
+            n_fg_class + 1,
+            roi_size=21, group_size=7,
+            spatial_scale=1. / self.feat_stride,
+            loc_normalize_mean=loc_normalize_mean,
+            loc_normalize_std=loc_normalize_std,
+            iter2=iter2)
+
+        mean = np.array([123.15, 115.90, 103.06],
+                        dtype=np.float32)[:, None, None]
+
+        super(FCISResNet101, self).__init__(
+            extractor, rpn, head,
+            mean, min_size, max_size,
+            loc_normalize_mean, loc_normalize_std)
+
+        if pretrained_model in self._models:
+            path = download_model(self._models[pretrained_model]['url'])
+            chainer.serializers.load_npz(path, self)
+
+
+class ResNet101Extractor(chainer.Chain):
+
+    def __init__(self):
```

Comment:
Can you change the feature extractor to take `initialW` as an argument, and pass the zero-initializer when a pretrained model is specified?

Reply:
OK